corehq/pillows/mappings/app_mapping.py

```diff
@@ -117,7 +117,7 @@
         "use_custom_suite": {"type": "boolean"},
         "cached_properties": {"type": "object", "dynamic": False},
         "modules": {
-            "type": "object",
+            "type": "nested",
```

Comment:
I do need to rewrite one query to support this, I think.  Working on that now

Reply:
https://github.com/dimagi/commcare-hq/pull/23079/commits/0f44e83eb97b202bbb55e75bd22d20940753b549